File: 20210523LangReview/Swift/02-ctor/ctor.cpp
/*

[ expriment result ]
===
- it turns out that C++ doesnt suppurt ctor inheritance
===
*/ 

#include <iostream>

class Base
{
private:
    int m_x;
public:
    Base(const int& x = 0)
    : m_x(x) {}

    virtual ~Base() {}
};

class Derived: public Baes {};

int main()
{
    Derived d1{1}; // error out; 
    
    return 0;
}

File: 20210523LangReview/Swift/02-ctor/ctor.cs
/*

[ expriment result ]
===
- it turns out that C# doesnt suppurt ctor inheritance
===
*/ 

using System;

namespace TS
{
    class Base
    {
        private int m_x;
        public Base(int x = 0)
        { this.m_x = x; }
    }

    class Derived: Base
    {}
    class TS
    {
        static void Main(string[] args)
        {
            Derived d1 = new Derived(1); // error out
            System.Console.WriteLine($"d1 {d1.ToString()}");

            Console.ReadLine();
        }
    }
}